spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/ScheduledTaskEndpoint.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.endpoint;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.scheduling.annotation.SchedulingConfigurer;
+import org.springframework.scheduling.config.CronTask;
+import org.springframework.scheduling.config.IntervalTask;
+import org.springframework.scheduling.config.ScheduledTaskRegistrar;
+import org.springframework.scheduling.config.Task;
+import org.springframework.scheduling.config.TriggerTask;
+import org.springframework.scheduling.support.ScheduledMethodRunnable;
+
+/**
+ * {@link Endpoint} to expose scheduled task information.
+ *
+ * @author Yunkun Huang
+ * @since 2.0.0
+ */
+@ConfigurationProperties(prefix = "endpoints.schedules")
+public class ScheduledTaskEndpoint
+		extends AbstractEndpoint<List<ScheduledTaskEndpoint.ScheduledTaskInformation>>
+		implements SchedulingConfigurer {
+	private List<ScheduledTaskInformation> list = new ArrayList<>();
+
+	public ScheduledTaskEndpoint() {
+		super("schedules");
+	}
+
+	/**
+	 * Create a new {@link ScheduledTaskEndpoint} instance.
+	 */
+	@Override
+	public List<ScheduledTaskInformation> invoke() {
+		return this.list;
+	}
+
+	@Override
+	public void configureTasks(ScheduledTaskRegistrar taskRegistrar) {
+		this.list.addAll(build(taskRegistrar.getFixedRateTaskList(), ScheduledType.FIXEDRATE));
+		this.list.addAll(build(taskRegistrar.getCronTaskList(), ScheduledType.CRON));
+		this.list.addAll(build(taskRegistrar.getFixedDelayTaskList(), ScheduledType.FIXEDDELAY));
+		this.list.addAll(build(taskRegistrar.getTriggerTaskList(), ScheduledType.TRIGGER));
+	}
+
+	private List<ScheduledTaskInformation> build(final List<? extends Task> tasks, final ScheduledType type) {
+		return tasks.stream().map(task -> {
```

Comment:
With so long method, usage of stream and map looks like of abusing such functionality IMHO.